tests/Unit/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsDataProviderTest.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\PrestaShopBundle\Form\Admin\Sell\Order\Invoices;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Form\ErrorMessage\ConfigurationErrorInterface;
+use PrestaShop\PrestaShop\Core\Form\ErrorMessage\InvoiceConfigurationError;
+use PrestaShopBundle\Form\Admin\Sell\Order\Invoices\InvoiceOptionsDataProvider;
+use PrestaShopBundle\Form\Admin\Sell\Order\Invoices\InvoiceOptionsType;
+use PrestaShopBundle\Form\Exception\DataProviderException;
+
+class InvoiceOptionsDataProviderTest extends TestCase
+{
+    /**
+     * Tests that exception is thrown if invoice number is incorrect
+     */
+    public function testValidateFailsOnIncorrectInvoiceNumber(): void
+    {
+        $dataConfiguration = $this->getMockBuilder(DataConfigurationInterface::class)->getMock();
+        $invoiceOptionsDataProvider = new InvoiceOptionsDataProvider($dataConfiguration, 5);
+        $data = [
+            InvoiceOptionsType::FIELD_INVOICE_NUMBER => 4,
+        ];
+        $exceptionThrown = false;
+        $configurationError = new InvoiceConfigurationError(InvoiceConfigurationError::ERROR_INCORRECT_INVOICE_NUMBER, InvoiceOptionsType::FIELD_INVOICE_NUMBER);
+        try {
```

Comment:
PHPUnit provides some assertions for exceptions. Did you choose not to use them on purpose?

Reply:
https://phpunit.readthedocs.io/en/latest/writing-tests-for-phpunit.html#writing-tests-for-phpunit-exceptions-examples-exceptiontest-php

Reply:
I will read the doc, but the reason I chose not to use it is that I want to check my exception for correct configuration errors. I am not sure if PHP unit supports that functionality.